### PR TITLE
Bundle PostCSS Global Data Plugin with default configuration

### LINKED ISSUE #340: RFC: Bundle PostCSS Global Data Plugin with default configuration
We are more and more drifting away from having one main CSS bundle in favor of having multiple separate CSS files. In the WordPress context individual blocks may have their own stylesheets or templates have their own ones. 

One issue that we manually have to solve on every project individually is how we share global stuff such as media queries, variables, mixins etc. across these different CSS entrypoints. 

The [`@csstools/postcss-global-data`](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-global-data) plugin aims to solve just that. We should bundle this with toolkit and allow you to place any global shared postcss config files in a specified location in the root of the project. All these files would then automatically get loaded before each entrypoint so that mixins, custom breakpoints, selectors etc. can be shared without having to manually import things into every single file. 

image

### PR DESCRIPTION
**Pull request by fabiankaegy**
Related Issue/RFC: #340

### Description of the Change

This change makes it possible to share one global configuration for things like custom breakpoints, selectors, mixins etc between all CSS entrypoints of a project. 

Especially with the introduction of block specific stylesheets one common issue is that the global elements such as `@custom-media` definitions are not shared across all these individual entrypoints. This leads to unintended breakages when an engineer uses these variables thinking they will work just as expected from the main CSS file. But in reality this causes the styles to be broken because they cannot read the mixin that is only defined in the other entrpoint.

With this change both the custom media etc definitions, and any mixins get a new location in the theme located at `./assets/css/globals/` & `./assets/css/mixins/`. All the `.css` files nested in that folder or any nested folders will now automatically get loaded and shared for all entrypoints. 

These folders are configurable at the project level via either the `paths.config.js` file or the `paths` key in the `10up-toolkit` object of `package.json` files. This flexibility to customize the paths also makes it possible to reference the folders of the theme in a plugin for example if you are developing a feature plugin that contains CSS and need to reference the themes media queries for example.

> [!NOTE]
> This will require a change in the wp-scaffold once this has been released. I've created a draft PR for it https://github.com/10up/wp-scaffold/pull/228 However, this is not a breaking change :) 

### Alternate Designs

Initially, I wanted the globals to live outside of the theme folder at the monorepo root level to make it more clear that these elements are shared across all instances. But I think this here with the ability to customize it is the better experience.

### Possible Drawbacks

This is a slight change in the development process that will need to be explained to the team. But it should be very nondisruptive. 

### Verification Process

The bundled sample theme project has been updated to use this new approach. Running the build in the theme directory will result in the mixins & globals getting applied both to the `dist/css/frontend.css` and the `dist/blocks/example-block/style.css` files.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.
- [x] I have added a changeset to my PR. See [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document for instructions

### CODE REVIEW

File: .changeset/thin-geckos-sparkle.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"10up-toolkit": minor
+"tenup-theme": minor
```
Comment (nicholasio) [inline .changeset/thin-geckos-sparkle.md]: No need to patch the tenup-theme, you can remove this line from this file.

File: packages/toolkit/config/postcss.config.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
 const path = require('path');
+const glob = require('fast-glob');
+
+const { getTenUpScriptsConfig } = require('../utils');
 
 module.exports = ({ file, env }) => {
+	const projectConfig = getTenUpScriptsConfig();
+	const { globalStylesDir, globalMixinsDir } = projectConfig.paths;
+
+	const globalCssFiles = glob.sync(`${globalStylesDir}**/*.css`);
```
Comment (nicholasio) [inline packages/toolkit/config/postcss.config.js:10]: Will this fail or just return an empty result if the global folder doesn't exist, or if it doesn't have any css files?

File: packages/toolkit/config/postcss.config.js [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
+	const globalMixinFiles = glob.sync(`${globalMixinsDir}**/*.css`);
+
 	const config = {
 		plugins: {
 			'postcss-import': {},
-			'postcss-mixins': {},
+			'@csstools/postcss-global-data': {
+				files: globalCssFiles,
```
Comment (nicholasio) [inline packages/toolkit/config/postcss.config.js:17]: What would files and mixins files be if there are no files? Should we only set this up if there are files?

File: packages/toolkit/config/postcss.config.js (hunk 2 above)
Comment (fabiankaegy) [reply packages/toolkit/config/postcss.config.js:10]: In my testing it always returned an empty array

File: packages/toolkit/config/postcss.config.js (hunk 3 above)
Comment (fabiankaegy) [reply packages/toolkit/config/postcss.config.js:17]: In my testing the value just is an empty array if there are no files. Which is the default value

File: packages/toolkit/__tests__/build-project-global-css/__fixtures__/assets/css/frontend/styles.css [hunk 4]
```diff
@@ -0,0 +1,10 @@
+html {
+	background: #f5f5f5;
+	padding: 20px;
+
+	@mixin margin-trim;
```
Comment (nicholasio) [inline packages/toolkit/__tests__/build-project-global-css/__fixtures__/assets/css/frontend/styles.css:5]: @fabiankaegy I set up this test and for some reason, this line breaks building this file! It doesn't even create the css file.

Did I miss something? I'm wondering if this mixins is undefined and therefore breaking everyting. On the other hand I'd like to understand why this test doesn't even generate the css file with this line and it also doesn't give any error.

To run tests locally just do `npm run test -w=packages/toolkit -- --watch` you can run the tests then inspect the dist folder for each test to see what toolkit is outputting

**Review by nicholasio — COMMENTED** [review]
@fabiankaegy I added a basic test but ran into a weird issue. Would love to figure it out before releasing a beta version of this.

File: packages/toolkit/__tests__/build-project-global-css/__fixtures__/assets/css/frontend/styles.css (hunk 4 above)
Comment (fabiankaegy) [reply packages/toolkit/__tests__/build-project-global-css/__fixtures__/assets/css/frontend/styles.css:5]: @nicholasio it appears that the issue was due to the `paths` not being set correctly for the `__fixtures__` path. See [52e5b67](https://github.com/10up/10up-toolkit/pull/389/commits/52e5b672a20209892a357b74a969cbbb1003bc89)

Comment (nicholasio) [reply packages/toolkit/__tests__/build-project-global-css/__fixtures__/assets/css/frontend/styles.css:5]: That makes sense! Thanks!

**Review by nicholasio — APPROVED** [review]

**Pull request merged**